### DISCUSSION TITLE
Warning using Union Types

### DISCUSSION THREAD

**Discussion by wsaca** [vue-macros/vue-macros#243] [Q&A] [answered]
Hi, I would like to know if this example is supported:

```ts
interface Transition {
  duration?: string;
}

export interface HelloWorldProps {
  msg?: string;
  transition?: Transition | boolean;
}

```

The console displays this warning:
```
[Vue warn]: Invalid prop: type check failed for prop "transition". Expected Null | Boolean, got Object 
```

Example:
https://stackblitz.com/edit/vitejs-vite-bcwnrm?file=src/components/types.ts

**Comment by sxzz** [comment, marked as answer]
It's a bug of vue core 
https://github.com/vuejs/core/issues/7511
